plugin/providers/keboola/resource_keboola_gooddata_user_management_v2.go

```diff
@@ -0,0 +1,279 @@
+package keboola
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/plmwong/terraform-provider-keboola/plugin/providers/keboola/buffer"
+	"net/url"
+)
+
+const userManagementComponentTemplate = "storage/components/kds-team.app-gd-user-management/configs/%s"
+
+type GoodDataUserManagementComponentV2 struct {
+	Name          string                                `json:"name"`
+	Description   string                                `json:"description"`
+	Configuration GoodDataUserManagementConfigurationV2 `json:"configuration"`
+}
+
+type GoodDataUserManagementConfigurationV2 struct {
+	Storage    GoodDataUserManagementStorageV2    `json:"storage"`
+	Parameters GoodDataUserManagementParametersV2 `json:"parameters"`
+}
+
+type GoodDataUserManagementParametersV2 struct {
+	Password     string `json:"#password"`
+	ProjectId    string `json:"pid"`
+	Username     string `json:"username"`
+	CustomDomain string `json:"domain_custom"`
+}
+
+type GoodDataUserManagementStorageV2 struct {
+	Input GoodDataUserManagementStorageInputV2 `json:"input"`
+}
+
+type GoodDataUserManagementStorageInputV2 struct {
+	Tables []GoodDataUserManagementInputTableV2 `json:"tables"`
+}
+
+type GoodDataUserManagementInputTableV2 struct {
+	Columns       []string `json:"columns"`
+	Source        string   `json:"source"`
+	Destination   string   `json:"destination"`
+	WhereColumn   string   `json:"where_column"`
+	WhereOperator string   `json:"where_operator"`
+	WhereValues   []string `json:"where_values"`
+}
+
+func resourceKeboolaGoodDataUserManagementAppV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGoodDataUserManagementCreateV2,
+		Read:   resourceGoodDataUserManagementReadV2,
+		Update: resourceGoodDataUserManagementUpdateV2,
+		Delete: resourceGoodDataUserManagementDeleteV2,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Required: true,
+				Type:     schema.TypeString,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"project_id": {
+				Required: true,
+				Type:     schema.TypeString,
+			},
+			"login": {
+				Required: true,
+				Type:     schema.TypeString,
+			},
+			"hashed_password": {
```

Comment:
It might be worth introducing a `ValidateFunc` on this property, to validate that the password has been hashed/encrypted and is not just a random string of text... I think there's `validateKBCEncryptedValue` used in the Snowflake / PostgreSQL resources that might be re-usable here.